app/controllers/application_controller.rb

```diff
@@ -15,10 +15,21 @@ class ApplicationController < ActionController::Base
   rescue_from ActiveRecord::RecordNotFound, with: :not_found!
 
   def current_organization
-    @current_organization ||= Organization.find_by(short_name: params[:organization_id]) || current_user&.organization
+    return @current_organization if @current_organization
```

Comment:
I'm not sure why you are no longer setting @current_organization if it isn't already set here now.